nvflare/edge/device_simulator/device_simulator.py

```diff
@@ -112,9 +109,7 @@ def fetch_task(self, job: JobResponse) -> Optional[TaskResponse]:
                 elif task.status == "DONE":
                     task["task_done"] = True
                     return task
-                elif task.status == "NO_TASK":
-                    return None
-                elif task.status == "RETRY":
+                elif task.status == "NO_TASK" or "RETRY":
```

Comment:
Are you sure about this statement? I think it always yield true, it should be,
    elif task.status in {"NO_TASK", "RETRY"}:

You should also use the constants here.